src/include/duckdb/common/vector_operations/generic_executor.hpp

```diff
@@ -93,28 +116,40 @@ template <class A_TYPE, class B_TYPE>
 struct StructTypeBinary {
 	A_TYPE a_val;
 	B_TYPE b_val;
+	bool a_is_null;
+	bool b_is_null;
 
 	using STRUCT_STATE = StructTypeState<2>;
 
-	static bool ConstructType(STRUCT_STATE &state, idx_t i, StructTypeBinary<A_TYPE, B_TYPE> &result) {
+	bool ContainsNull() {
+		return a_is_null || b_is_null;
```

Comment:
Note that nulls can be at different levels - i.e. the struct itself can be NULL, as well as the child elements. This should also check if the struct itself is `NULL` (so `is_null || a_val.is_null || b_val.is_null`)